src/shared/components/home/home.tsx

```diff
@@ -791,6 +801,25 @@ export class Home extends Component<HomeRouteProps, HomeState> {
             onChange={this.handleDataTypeChange}
           />
         </div>
+        {dataType === DataType.Post &&
+          (this.state.siteRes.my_user || UserService.Instance.myUserInfo) && (
+            <div className="col-auto form-check form-switch">
```

Comment:
Since this is used in two places, externalize it into its own component.

Also, this is causing a lot of wrapping even on desktop, so we need to figure out how to make it skinnier, as well as more uniform with the rest of these toggles/filters.

How about doing it like all the other buttons, with the two options being `eye-on` and `eye-off` icons. Check out src/shared/components/common/data-type-select.tsx for an example. 

And to explain what it does on a click/touch and hold, use `data-tippy-content`.

Reply:
> Since this is used in two places, externalize it into its own component.

I started doing that and am now questioning the usefulness of that. When I tried, I ended up with a component with just the input and label taking `showHidden` and `onShowHiddenChangeProps`. I do think a case can be made for extracting the entire filter/sort bar into it's own component. That would allow us to have a single source of truth for this instead of repeating logic on the homepage and community page.

> How about doing it like all the other buttons, with the two options being eye-on and eye-off icons. Check out src/shared/components/common/data-type-select.tsx for an example.

Do you mean the pillbox looking button groups? Like this but with eye and eye slash instead of posts and comments?
![image](https://github.com/LemmyNet/lemmy-ui/assets/28871516/5e2dc52a-6311-487c-be46-79c6bc3f738a)

Reply:
> I started doing that and am now questioning the usefulness of that. When I tried, I ended up with a component with just the input and label taking showHidden and onShowHiddenChangeProps

Sure, but then you only need to change markup / visual related things in one place. That's why we have the listingTypeSelect and SortTypeSelect as their own components too, even though they are pretty simple.

> I do think a case can be made for extracting the entire filter/sort bar into it's own component.

Maybe, although the indivdual Select components are used elsewhere besides just the Home and Community pages iirc. Either way that could be a separate PR as it would be a bigger change. Plus even the Home and Community have different filters (no subscribed vs local vs all filter).

> Do you mean the pillbox looking button groups? Like this but with eye and eye slash instead of posts and comments?

Yep. Just to look uniform with the others.